gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/language/translator/DotNetTranslateVisitor.java

```diff
@@ -155,10 +156,10 @@ public Void visitFloatLiteral(final GremlinParser.FloatLiteralContext ctx) {
     public Void visitDateLiteral(final GremlinParser.DateLiteralContext ctx) {
         // child at 2 is the date argument to datetime() and comes enclosed in quotes
         final String dtString = ctx.getChild(2).getText();
-        final Date dt = DatetimeHelper.parse(removeFirstAndLastCharacters(dtString));
-        sb.append("DateTimeOffset.FromUnixTimeMilliseconds(");
-        sb.append(dt.getTime());
-        sb.append(")");
+        final OffsetDateTime dt = DatetimeHelper.parse(removeFirstAndLastCharacters(dtString));
```

Comment:
I think this is the wrong type name for .net. If it is indeed the wrong type, I'm wondering if we have sufficient test coverage for these translators.
```suggestion
        final DateTimeOffset dt = DatetimeHelper.parse(removeFirstAndLastCharacters(dtString));
```